authors/apps/articles/views.py

```diff
@@ -0,0 +1,88 @@
+import uuid
+import jwt
+from django.shortcuts import render
+from django.template.defaultfilters import slugify
+from .renderers import ArticleJSONRenderer, ListArticlesJSONRenderer
+from rest_framework import status, exceptions
+from rest_framework.exceptions import PermissionDenied
+from rest_framework.response import Response
+from rest_framework.generics import get_object_or_404
+from rest_framework import generics, status
+from .models import Article
+from authors import settings
+
+
+from rest_framework.permissions import (IsAuthenticated,
+                                        IsAuthenticatedOrReadOnly)
+from .serializers import (
+    CreateArticleAPIViewSerializer
+)
+from authors.settings import RPD
+from ..authentication.models import User
+from .models import Article
+from rest_framework import generics
+
+
+class CreateArticleAPIView(generics.ListCreateAPIView):
+    permission_classes = (IsAuthenticated,)
+    queryset = Article.objects.all()
+    serializer_class = CreateArticleAPIViewSerializer
+
+    def post(self, request):
+        """ Method For Posting Article """
+        article = request.data.get('article', {})
+
+        serializer = self.serializer_class(data=article)
+        serializer.is_valid(raise_exception=True)
+        serializer.save(author=request.user)
+        print(serializer.data)
+
+        response = serializer.data
+        response.update(article)
+
+        return Response(response, status=status.HTTP_201_CREATED)
+
+
+class RetrieveArticleAPIView(generics.RetrieveUpdateDestroyAPIView):
+    permission_classes = (IsAuthenticated,)
+    serializer_class = CreateArticleAPIViewSerializer
+    queryset = Article.objects.all()
+
+    def retrieve(self, request, slug):
+        article = Article.objects.get(slug=slug)
+        serializer = self.serializer_class(article)
+        return Response(serializer.data)
+
+    def update(self, request, slug):
+        """This method updates a user article"""
+
+        article = request.data.get('article', {})
+        article["author"] = request.user.email
+
+        article_instance = Article.objects.get(slug=slug)
+        print(article_instance.author)
+        slug = slugify(article["title"]).replace("_", "-")
+        slug = slug + "-" + str(uuid.uuid4()).split("-")[-1]
+        article["slug"] = slug
+
+        if article_instance.author != request.user:
```

Comment:
Wouldn't it be better to first check user permission before you slugify the article title
```suggestion
        if article_instance.author != request.user:
              raise PermissionDenied
        
        slug = slugify(article["title"]).replace("_", "-")
        slug = slug + "-" + str(uuid.uuid4()).split("-")[-1]
        article["slug"] = slug
```

Reply:
Please humbly read and understand the flow of that entire method. Your current interpretation is wrong.
The If statement you are talking of checks whether the 'author' of the article is the same as the currently logged in user.

Reply:
Understood